test/autotest.py

```diff
@@ -15,6 +15,19 @@
 import stat
 import re
 
+
+# FIX for bad path for Java:  Travis prepended
+#     "/usr/bin:/opt/pyenv/libexec:/opt/pyenv/plugins/python-build/bin:/"
+# to os.environ['PATH'] on July 31, 2019.  It does this, even though
+# "/usr/bin" occurs later in the path.  /usr/bin/java exists as Java-8.
+# So, it never finds java (version Java-11) in
+#   "/usr/local/lib/jvm/openjdk11/bin", which occurs later in path.
+# Unfortunately, 'Makefile' finds javac as verion Javac-11,
+# and so the 'java1' test fails, using java (Java-8).
+if os.environ.has_key('PATH') and os.environ['PATH'].startswith('/usr/bin:') \
```

Comment:
Instead of having the PATH updates here, let's have configure detect and export a couple of make variables like JAVAC_PATH and JAVA_PATH that we can use directly in the test suite. That way we won't have any side-effects here.

Reply:
@karya0, It won't work to configure for JAVA_PATH and JAVAC_PATH.  We have to test for this inside Python when hitting the Travis bug.

Basically, we're doing a hack-ish workaround because Travis si broken here.  For reasons that I don't understand completely, Travis decides to use pyenv for some PR's, but not others.  In this case, Travis wants to use pyenv for Python.  So, Travis prepends to the PATH and places /usr/bin earliy int he path, which is incorrect.  So, in Python _only_, we see a PATH that causes us to use /usr/bin/java (which is wrong, it's java-v8) instead of the correct path to a java v11.  This occurs _only_ inside Python due to some weird thing done _only_ by Travis.